src/okta/types/cache.ts

```diff
@@ -0,0 +1,49 @@
+import { OktaApplication, OktaFactor, OktaUser, OktaUserGroup } from '.';
+import { OktaApplicationGroup, OktaApplicationUser } from './applications';
+
+export interface OktaUserCacheData {
+  user: OktaUser;
+  factors: OktaFactor[];
+  userGroups: OktaUserGroup[];
+}
+
+export interface OktaUserCacheEntry {
+  key: string;
+  data?: OktaUserCacheData;
+}
+
+export interface OktaApplicationCacheData {
+  application: OktaApplication;
+  applicationGroups: OktaApplicationGroup[];
+}
+
+export interface OktaApplicationCacheEntry {
+  key: string;
+  data?: OktaApplicationCacheData;
+}
+
+export interface OktaApplicationUserCacheData {
+  applicationId: string;
+  applicationUser: OktaApplicationUser;
+}
+
+export interface OktaApplicationUserCacheEntry {
+  key: string;
+  data?: OktaApplicationUserCacheData;
+}
+
+export interface OktaCacheState {
+  /**
+   * Number of resources seen from resource API.
+   */
+  seen: number;
+
+  /**
+   * Number of keys reported by `cache.putEntries()`.
+   */
+  putEntriesKeys: number;
+
+  fetchCompleted: boolean;
+
+  encounteredAuthorizationError?: boolean;
+}
```

Comment:
These types are used to reflect the data that was being tracked across invocations of the Lambda during iterative fetching. None of these types should need to exist in the new integration, so I think this PR can exclude this stuff.

Reply:
Removed